ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Connection.scala

```diff
@@ -192,12 +192,15 @@ private[h2] class H2Connection[F[_]](
     firstGoAway.getOrElse(F.unit) >> go(chunk)
   }
 
-  def writeLoop: Stream[F, Nothing] =
-    Stream
-      .fromQueueUnterminatedChunk[F, H2Frame](outgoing, Int.MaxValue)
-      .chunks
-      .foreach(writeChunk)
```

Comment:
Oh you're right, the existing code has the same issues, good catch ...

Reply:
Given that the existing code has this issue, do we still wanna block this on determining and using a limit?

Reply:
I have just written 16. Perhaps it should be configurable, in next PR, but it is a bit of an internal detail. As it is (with the hardcoded limit), it's better that what we have ow.

An optimal solution, it occurs to me, would be to add a `foreach` method to the Queue itself in `cats-effect`., given that it is a common pattern to enqueue to drain, without using a stream in between.